packages/@uppy/aws-s3-multipart/types/index.d.ts

```diff
@@ -22,7 +22,7 @@ export interface AwsS3MultipartOptions extends PluginOptions {
     ) => MaybePromise<AwsS3Part[]>
     prepareUploadParts?: (
       file: UppyFile,
-      partData: { uploadId: string; key: string; partNumbers: Array<number>; chunks: { [k: number]: Blob } }
+      partData: { uploadId: string; key: string; parts: Record<number, Blob> }
```

Comment:
my understanding of type [Record<K, T>](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type) implies that `K` is the type of the keys expected, while `T` is the type of values expected.

So `Record<number, Blob>` would expect something like:
```ts
const parts = {1: new Blob('foo'), 4: new Blob('bar)}
```

Is that the intended type?

My thought is that we want something along the lines of `Array<{number: number; chunk: Blob}>` instead.

Reply:
Good catch!